src/main/java/goral/psychotherapistoffice/web/visitor/VisitorController.java

```diff
@@ -37,41 +39,55 @@ public VisitorController(PatientService patientService, CalenderService calender
 
     //stała z UserManagementController
     public static final String NOTIFICATION_ATTRIBUTE = "visitorNotification";
+    public static final String NOTIFICATION_ATTRIBUTE2 = "visitorNotification2";
 
     @GetMapping("/visitor")
     public String getAdminPanel(){
         return "visitor/visitor";
     }
 
 
-    @GetMapping("/visitor/dadaj-rezerwacje")
-    public String addMeetingForm(Model model){
 
-
-        List<TherapyDto>allTherapies = therapyService.findAllTherapies();
+    @GetMapping("/termin/{id}")
+    public String addMeetingByVisitor(@PathVariable long id, Model model){
+        //1. wybranie z kalendarza
+        CalenderDto calender = calenderService.findCalenderByIdIfFreeIsTrue(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+        model.addAttribute("calender", calender);
+        PatientDto patientDto = new PatientDto();
+        model.addAttribute("patientDto", patientDto);
+        List<TherapyDto> allTherapies = therapyService.findAllTherapies();
         model.addAttribute("therapies", allTherapies);
-        List<PatientDto>allPatients = patientService.findAllPatients();
-        model.addAttribute("patients", allPatients);
-        List<CalenderDto> allFreeTherms = calenderService.findAllFreeTherms();
-        model.addAttribute("therms", allFreeTherms);
+        //3. ustawienie meetingu
         MeetingVisitorSaveDto meetingVisitorSaveDto = new MeetingVisitorSaveDto();
         model.addAttribute( "meetingSave", meetingVisitorSaveDto);
         return "visitor/meeting-form";
-
     }
 
-    @PostMapping("/visitor/dadaj-rezerwacje")
-    public String addMeeting(MeetingVisitorSaveDto meetingSave, RedirectAttributes redirectAttributes) {
-        meetingService.addMeeting(meetingSave);
-
+    @PostMapping("/termin/{id}")
+    public String addPatientAndMeeting(@ModelAttribute("patientDto") PatientDto patientDto, @ModelAttribute("meetingSave") MeetingVisitorSaveDto meetingSave, RedirectAttributes redirectAttributes){
+        patientService.addPatient(patientDto);
+        meetingSave.setPatient(patientDto.getId());
```

Comment:
I am thinking about this... we have in this exception, i am right?

Reply:
```java
    @Transactional
    public void addPatient(PatientDto patientDto){
        Patient patientToSave = new Patient();
        patientToSave.setNick(patientDto.getNick());
        patientToSave.setName(patientDto.getName());
        patientToSave.setSurname(patientDto.getSurname());
        patientToSave.setTelephone(patientDto.getTelephone());
        patientToSave.setYearOfBrith(patientDto.getYearOfBrith());
        patientRepository.save(patientToSave);

    }
```
so, when we run this method `patientRepository.save(patientToSave);` maybe entity has id value. in my symfony technology when we save entity in database then automatically it is getting id value

you can try this
```java
    @Transactional
    public PatientDto addPatient(PatientDto patientDto){
        Patient patientToSave = new Patient();
        patientToSave.setNick(patientDto.getNick());
        patientToSave.setName(patientDto.getName());
        patientToSave.setSurname(patientDto.getSurname());
        patientToSave.setTelephone(patientDto.getTelephone());
        patientToSave.setYearOfBrith(patientDto.getYearOfBrith());
        patientRepository.save(patientToSave);

        return patientToSave;
    }
    ```
    
and in controller
    
you can try replace this
```java
patientService.addPatient(patientDto);
meetingSave.setPatient(patientDto.getId());
```

for this code

```java
meetingSave.setPatient(patientService.addPatient(patientDto));
```